File: interviewbit/math/reverse-integer.java
/*
    interviewbit
    track: "programming",
    section: "math",
    problem title: "Reverse Integer"
*/

import java.lang.Math;
public class Solution {
    public int reverse(int A) {
        if(A<9 && A>-9){
            return A;
        }
        boolean negFlag = A<0;
        if(negFlag){
            A = -1*A;
        }
        int digitLength = 1;
        while(Math.floor((A/Math.pow(10,digitLength)))!=0)digitLength++;
        // System.out.println(digitLength);
        
        long result = 0;
        for(int i=digitLength-1,j=0;i>=0;i--,j++){
            int currentDigit = ((int)(A/Math.pow(10,i)))%10;
            result+=(long)(currentDigit*Math.pow(10,j));
            if(result>Integer.MAX_VALUE){
                return 0;
            }
        }
        if(negFlag)result = -1*result;
        return (int)result;
        
        
    }
}

File: interviewbit/math/reverse-integer.js
/*
    interviewbit
    track: "programming",
    section: "math",
    problem title: "Reverse Integer"
*/

function get32BitInteger(){
    var value = 0;
    //counting only 31 bits because it's a signed int
    for(var i=0;i<31;i++){
        value+=Math.pow(2,i);
    }
    return value;
}

module.exports = { 
 //param A : integer
 //return an integer
	reverse : function(A){
        var MAX_VAL = get32BitInteger();
        var sign = A<0 ? -1 : 1;
        var reverse_value = parseInt(A.toString().split('').reverse().join(''))
        if((reverse_value)>MAX_VAL){
            return 0;
        }
        return sign*reverse_value;
	}
};
